tests/verify_MPI_not_initialized.py

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python
+
+"""
+Some HPC systems only allow MPI to be initialized when jobs are
+submitted through slurm, not on login nodes.  To make the non-MPI
+parts of simsopt usable on login nodes, then, we want to be sure that
+simsopt does not initialize MPI (via mpi4py) unless classes are
+imported that require MPI, such as MpiPartition or Vmec. This script
+checks to be sure that importing the top-level simsopt module and
+importing a simsopt geometry class do not initialize MPI.
+
+We must do this check in an isolated script rather than in the set of
+unit tests, because otherwise one of the unit tests that uses MPI
+would initialize MPI.
+"""
+
+import simsopt
+from simsopt.geo.surfacerzfourier import SurfaceRZFourier
+
+import mpi4py
+# The next line tells mpi4py to not initialize MPI when mpi4py.MPI is imported:
+mpi4py.rc.initialize = False
+
+from mpi4py import MPI
+
+if mpi4py.MPI.Is_initialized():
+    print("ERROR! Importing simsopt should not initialize MPI")
+    exit(1)
+else:
+    print("Verified that importing simsopt does not initialize MPI")
```

Comment:
Instead of this complicated setup, a simple approach would be to check for mpi4py in loaded modules. Top level import of simsopt shouldn't load mpi4py. This can be checked with these few lines of code
```
import sys
assert("mpi4py" not in sys.modules)
```

Lines 20  to end can be replaced with the above two lines.

Reply:
Here's the reason the new test was written that way. `FiniteDifference` and `MpiFiniteDifference` are in the same module, with the serial `FiniteDifference` imported when the top-level simsopt module is imported via `least_squares_serial_solve`. So the `@simsoptRequires` in front of `MpiFiniteDifference` requires that simsopt tries to import mpi4py when a user calls `import simsopt` or imports the serial `FiniteDifference`. Importing `mpi4py` does not initialize MPI; usually importing `mpi4py.MPI` does, but not if `mpi4py.rc.initialize=False`. So checking `"mpi4py" in sys.modules` does not generally tell us whether MPI has been initialized. Whereas `mpi4py.MPI.Is_initialized()` gives the answer reliably. In `finite_difference.py` and perhaps elsewhere in the future, it can be useful to `try: import mpi4py` to see if mpi4py has been installed in a module that we still may wish to use on a login node, i.e. without initializing MPI.